components/director/internal/open_resource_discovery/service.go

```diff
@@ -0,0 +1,529 @@
+		}
+	}
+
+	return s.tombstoneSvc.ListByApplicationID(ctx, appID)
+}
+
+func (s *Service) resyncPackage(ctx context.Context, appID string, packagesFromDB []*model.Package, pkg model.PackageInput) error {
+	ctx = addFieldToLogger(ctx, "package_ord_id", pkg.OrdID)
+	if i, found := searchInSlice(len(packagesFromDB), func(i int) bool {
+		return packagesFromDB[i].OrdID == pkg.OrdID
+	}); found {
+		return s.packageSvc.Update(ctx, packagesFromDB[i].ID, pkg)
+	}
+	_, err := s.packageSvc.Create(ctx, appID, pkg)
+	return err
+}
+
+func (s *Service) resyncBundle(ctx context.Context, appID string, bundlesFromDB []*model.Bundle, bndl model.BundleCreateInput) error {
+	ctx = addFieldToLogger(ctx, "bundle_ord_id", *bndl.OrdID)
+	if i, found := searchInSlice(len(bundlesFromDB), func(i int) bool {
+		return equalStrings(bundlesFromDB[i].OrdID, bndl.OrdID)
+	}); found {
+		return s.bundleSvc.Update(ctx, bundlesFromDB[i].ID, bundleUpdateInputFromCreateInput(bndl))
+	}
+	_, err := s.bundleSvc.Create(ctx, appID, bndl)
+	return err
+}
+
+func (s *Service) resyncProduct(ctx context.Context, appID string, productsFromDB []*model.Product, product model.ProductInput) error {
+	ctx = addFieldToLogger(ctx, "product_ord_id", product.OrdID)
+	if i, found := searchInSlice(len(productsFromDB), func(i int) bool {
+		return productsFromDB[i].OrdID == product.OrdID
+	}); found {
+		return s.productSvc.Update(ctx, productsFromDB[i].OrdID, product)
+	}
+	_, err := s.productSvc.Create(ctx, appID, product)
+	return err
+}
+
+func (s *Service) resyncVendor(ctx context.Context, appID string, vendorsFromDB []*model.Vendor, vendor model.VendorInput) error {
+	ctx = addFieldToLogger(ctx, "vendor_ord_id", vendor.OrdID)
+	if i, found := searchInSlice(len(vendorsFromDB), func(i int) bool {
+		return vendorsFromDB[i].OrdID == vendor.OrdID
+	}); found {
+		return s.vendorSvc.Update(ctx, vendorsFromDB[i].OrdID, vendor)
+	}
+	_, err := s.vendorSvc.Create(ctx, appID, vendor)
+	return err
+}
+
+func (s *Service) resyncAPI(ctx context.Context, appID string, apisFromDB []*model.APIDefinition, bundlesFromDB []*model.Bundle, packagesFromDB []*model.Package, api model.APIDefinitionInput) error {
+	ctx = addFieldToLogger(ctx, "api_ord_id", *api.OrdID)
+	i, found := searchInSlice(len(apisFromDB), func(i int) bool {
+		return equalStrings(apisFromDB[i].OrdID, api.OrdID)
+	})
+
+	var bundleID *string
+	var packageID *string
+
+	if i, found := searchInSlice(len(bundlesFromDB), func(i int) bool {
+		return equalStrings(bundlesFromDB[i].OrdID, api.OrdBundleID)
+	}); found {
+		bundleID = &bundlesFromDB[i].ID
+	}
+
+	if i, found := searchInSlice(len(packagesFromDB), func(i int) bool {
+		return equalStrings(&packagesFromDB[i].OrdID, api.OrdPackageID)
+	}); found {
+		packageID = &packagesFromDB[i].ID
+	}
+
+	specs := make([]*model.SpecInput, 0, len(api.ResourceDefinitions))
+	for _, resourceDef := range api.ResourceDefinitions {
+		specs = append(specs, resourceDef.ToSpec())
+	}
+
+	if !found {
```

Comment:
Perhaps rename the variable `found` to something else like `isAPIFound` because currently it is a bit puzzling what `found` is referring to as it is referring to found on line 391